From: Laurent Bigonville <bigon@debian.org>
Date: Sun, 19 Mar 2023 12:15:50 +0100
Subject: Fix FTBFS on x32

This patch is inspired from Simon proposal on the upstream bug:
https://gitlab.gnome.org/GNOME/gcr/-/issues/45
---
 egg/egg-asn1x.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/egg/egg-asn1x.c b/egg/egg-asn1x.c
index b7d9d11..63e6b16 100644
--- a/egg/egg-asn1x.c
+++ b/egg/egg-asn1x.c
@@ -2213,7 +2213,7 @@ anode_read_time (GNode *node,
 		return anode_failure (node, "invalid time content");
 
 	/* In order to work with 32 bit time_t. */
-	if (sizeof (time_t) <= 4 && when->tm_year >= 138) {
+	if ((sizeof (time_t) <= 4 || sizeof (glong) <= 4) && when->tm_year >= 138) {
 		*value = (time_t)2145914603;  /* 2037-12-31 23:23:23 */
 
 	/* Convert to seconds since epoch */
@@ -4740,7 +4740,7 @@ egg_asn1x_parse_time_general (const gchar *time, gssize n_time)
 		return -1;
 
 	/* In order to work with 32 bit time_t. */
-	if (sizeof (time_t) <= 4 && when.tm_year >= 138) {
+	if ((sizeof (time_t) <= 4 || sizeof (glong) <= 4) && when.tm_year >= 138) {
 		value = (time_t)2145914603;  /* 2037-12-31 23:23:23 */
 
 	/* Convert to seconds since epoch */
@@ -4771,7 +4771,7 @@ egg_asn1x_parse_time_utc (const gchar *time, gssize n_time)
 		return -1;
 
 	/* In order to work with 32 bit time_t. */
-	if (sizeof (time_t) <= 4 && when.tm_year >= 138) {
+	if ((sizeof (time_t) <= 4 || sizeof (glong) <= 4) && when.tm_year >= 138) {
 		value = (time_t)2145914603;  /* 2037-12-31 23:23:23 */
 
 	/* Convert to seconds since epoch */
